plugin/mode_indicator/mode_indicator.py

```diff
@@ -0,0 +1,149 @@
+from talon import Module, app, registry, scope, ui
+from talon.canvas import Canvas
+from talon.screen import Screen
+from talon.skia.canvas import Canvas as SkiaCanvas
+from talon.skia.imagefilter import ImageFilter
+from talon.ui import Rect
+
+canvas: Canvas = None
+current_mode = ""
+mod = Module()
+
+setting_show = mod.setting(
+    "mode_indicator_show",
+    bool,
+    desc="If true the mode indicator is shown",
+)
+setting_size = mod.setting(
+    "mode_indicator_size",
+    float,
+    desc="Mode indicator diameter in pixels",
+)
+setting_x = mod.setting(
+    "mode_indicator_x",
+    float,
+    desc="Mode indicator center X-position in percentages(0-1). 0=left, 1=right",
+)
+setting_y = mod.setting(
+    "mode_indicator_y",
+    float,
+    desc="Mode indicator center Y-position in percentages(0-1). 0=top, 1=bottom",
+)
+setting_color_alpha = mod.setting(
+    "mode_indicator_color_alpha",
+    float,
+    desc="Mode indicator alpha/opacity in percentages(0-1). 0=fully transparent, 1=fully opaque",
+)
+setting_color_sleep = mod.setting("mode_indicator_color_sleep", str)
+setting_color_dictation = mod.setting("mode_indicator_color_dictation", str)
+setting_color_mixed = mod.setting("mode_indicator_color_mixed", str)
+setting_color_other = mod.setting("mode_indicator_color_other", str)
```

Comment:
Was kind of expecting this to be called "command", but I guess your name is more accurate?

Reply:
Yeah the logic is implemented that this is the else condition. For example if you have a game mode or something else. Without that piece of logic the mode indicator would have an undefined behavior for those kind of scenarios. The comment in the Talon file above that setting mentions command mode.